src/lib/business/deals/__tests__/trendDetection.test.ts

```
@@ -0,0 +1,357 @@
+import { Deal } from '../../lib/entities/deals/Deal';
+
+// Mock current date for consistent testing
+const mockDate = new Date('2024-12-15T10:00:00Z');
+
+// Mock the Date constructor to return our fixed date
+const originalDate = global.Date;
+beforeAll(() => {
+  global.Date = class extends Date {
+    constructor(...args: any[]) {
+      if (args.length === 0) {
+        return mockDate;
```

Biome (1.9.4)

[error] The constructor should not return a value.

The constructor is here:

Returning a value from a constructor is ignored.

(lint/correctness/noConstructorReturn)